src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -393,7 +373,12 @@ public void updateOdometry() {
 
     // Updates pose based on vision
     if (m_visionEnabled) {
-      //TODO: there should be a cleaner way to prosses vision
+
+      // If the cosine of the picth or roll is less than 1, it is not 0
+      // The cosine should be less than 0.995 (5.73 degrees) if it goes over the charge station
+      if(getPitch().getCos()<0.995 || getRoll().getCos()<0.995){
```

Comment:
Probably more clear to just check the angle itself rather than the cosine of the angle. Also, check spelling in the comments